module4/homework/appendNewRecipe/AppendNewRecipe.cpp

```diff
@@ -0,0 +1,65 @@
+#include "AppendNewRecipe.hpp"
+
+#include <algorithm>
+#include <fstream>
+#include <map>
+
+static const std::map<char, std::string> units{{'g', "gram"},
+                                               {'s', "szklanka(i)"},
+                                               {'m', "mililitrow"}};
+
+bool AppendNewRecipe(const Steps& steps,
+                     const Ingredients& ingredients,
+                     const Amount& amount) 
+{
+    const std::string filename{"recipes.txt"};
+    std::fstream output(filename, output.out | output.app);
+    if (!output.is_open()) {
+        return false;
+    }
+    output << FormatRecipit(steps, ingredients, amount).str();
+    output.close();
+    return true;
+}
+
+FormattedIngredients FormatIngredients(const Ingredients& ingredients,
+                                       const Amount& amount)
+{
+    Steps result;
+    result.reserve(ingredients.size());
+    auto amount_it = amount.begin();
+
+    std::for_each(ingredients.begin(), ingredients.end(), 
+                    [&result, &amount_it](const auto& ingr){
+                        std::stringstream preparedIngredient{};
+                        preparedIngredient << (*amount_it).first << " "
+                                           << units.at((*amount_it).second) << " "
+                                           << ingr;
+                        result.push_back(preparedIngredient.str());
+                        ++amount_it;
+    });
```

Comment:
You use .at() here. You can get exception std::out_of_range. You can handle it here.

Reply:
I think transform will be better here. You can resize your vector or use back_inserter iterator.